src/gordon_gcp/plugins/event_consumer.py

```diff
@@ -188,21 +192,68 @@ def _init_subscriber_client(self, auth_client):
             logging.error(msg, exc_info=e)
             raise exceptions.GCPGordonError(msg)
 
-        return client
+        return client.subscribe(self.config['subscription'])
 
     def build_event_consumer(self):
         self._validate_config()
         validator = validate.MessageValidator()
         parser = parse.MessageParser()
         auth_client = self._init_auth()
-        subscriber = self._init_subscriber_client(auth_client)
+        subscription = self._init_subscriber_client(auth_client)
         loop = self.kwargs.get('loop', asyncio.get_event_loop())
 
         return GPSEventConsumer(
-            self.config, subscriber, validator, parser, self.success_channel,
+            self.config, subscription, validator, parser, self.success_channel,
             self.error_channel, loop)
 
 
+class _GPSThread(threading.Thread):
+    """Pub/Sub Message-specific thread with its own asyncio event loop.
+
+    .. attention::
+        This thread class is not meant to be worked with directly.
+
+    Google's Pub/Sub library is blocking. This thread class allows us
+    to create work in another thread/loop and not block the main event
+    loop within Gordon.
+
+    Inspiration directly from https://stackoverflow.com/a/29302684
+
+    Args:
+        event (threading.Event): event for thread instance to await
+            signal (via ``event.wait()``) to start work.
+        name (str): thread name
+        daemon (bool): whether the thread should be daemonized or not.
+            Defaults to ``False``.
+    """
+    def __init__(self, event, name=None, daemon=False):
+        super().__init__(name=name, daemon=daemon)
+        self.loop = None
+        self.thread_id = None
+        self.event = event
+
+    def run(self):
+        """Create and start thread's own event loop."""
+        self.loop = asyncio.new_event_loop()
+        self.thread_id = threading.current_thread()
+        self.loop.call_soon(self.event.set)
+        self.loop.run_forever()
+
+    def stop(self):
+        """Stop the thread's event loop. Thread safe."""
+        self.loop.call_soon_threadsafe(self.loop.stop)
+
+    def add_task(self, coro):
+        """Schedule a task to the thread's event loop. Thread safe.
+
+        Args:
+            coro: an asyncio `coroutine <https://docs.python.org/3/
+                library/asyncio-task.html#coroutine>`_ function
+        """
+        f = functools.partial(asyncio.ensure_future, coro, loop=self.loop)
+        self.loop.call_soon_threadsafe(f)
```

Comment:
Before, there was logic about whether the task was going to be called from within the thread or outside. I don't think there's an instance we're `gpsthread.add_task` would _not_ be called from outside the thread. Even if it is, I'm pretty sure it should be fine 🤔